edb/errors/__init__.py

```diff
@@ -152,6 +152,10 @@ class InvalidPropertyTargetError(InvalidTargetError):
     _code = 0x_04_02_01_02
 
 
+class InvalidScopeConfigurationError(InvalidTypeError):
```

Comment:
Probably not worth an explicit public error class, let's use `InvalidReferenceError` instead.

Reply:
Also we don't manually add types to this file. That's done via modifying `edb/api/errors.txt` and running `edb gen-errors`, as the comment on top of the file suggests.

> Probably not worth an explicit public error class, let's use InvalidReferenceError instead.

Maybe call it `ScopeError` and add it?

Reply:
I see no point in being hyper-granular in query error classes.  `InvalidReferenceError` is currently raised for cases where you refer to something that doesn't exist, or exists and is of a wrong type. In this case, the object you're referencing exists, but you can't refer to it in that scope.